### ISSUE TITLE
"Unknown disk type" error when specifying template_id and image_id of disk

### ISSUE THREAD

**Issue by ahmadalli** [OpenNebula/terraform-provider-opennebula#133]
I've been trying to prevent #97 using this resource:

```hcl
resource "opennebula_virtual_machine" "test" {
  name = "test"
  template_id = 0
  cpu = 2
  vcpu = 2
  memory = 2048

  disk {
    image_id = 0
    size = 10000
  }

  nic {
    network_id = 0
  }
}
```

but I'm getting this error:

> Error allocating a new virtual machine template. DISK 0: Unknown disk type

I've also tried setting `target` and `driver` but it didn't fix it

**Comment by ahmadalli** [comment]
my current workaround is I apply this template (I remove `image_id` from disk):

```hcl
resource "opennebula_virtual_machine" "test" {
  name = "test"
  template_id = 0
  cpu = 2
  vcpu = 2
  memory = 2048

  disk {
    size = 10000
  }

  nic {
    network_id = 0
  }
}
```

then I add `image_id` to it and run `terraform apply` again. Weirdly it'd update the `size` of image (from 0 to 10000) and then other `terraform apply` wouldn't result in any change

**Comment by ahmadalli** [comment]
this has been resolved in version 0.4.0
